src/rules/pylint/rules/too_many_statements.rs

```diff
@@ -0,0 +1,412 @@
+use crate::ast::helpers::identifier_range;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{ExcepthandlerKind, Stmt, StmtKind};
+
+define_violation!(
+    pub struct TooManyStatements {
+        pub statements: usize,
+        pub max_statements: usize,
+    }
+);
+
+impl Violation for TooManyStatements {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyStatements {
+            statements,
+            max_statements,
+        } = self;
+        format!("Too many statements ({statements}/{max_statements})")
+    }
+}
+
+fn get_num_statements(stmts: &[Stmt]) -> usize {
+    let mut count: usize = 0;
+    for stmt in stmts {
+        // TODO: Match will need be added once it's in ast.rs
+        match &stmt.node {
+            StmtKind::If { body, orelse, .. } => {
+                count += 1;
+                count += get_num_statements(body);
+                if !orelse.is_empty() {
+                    // else:
+                    //   if:
+                    // same as elif:
+                    // but otherwise else: counts as its own statement
+                    // necessary to avoid counting else and if separately when they appear after each other
+                    match orelse.first().unwrap().node {
+                        StmtKind::If { .. } => {}
+                        _ => {
+                            count += 1;
+                        }
+                    }
+                    count += get_num_statements(orelse);
+                }
+            }
+            StmtKind::For { body, orelse, .. } | StmtKind::AsyncFor { body, orelse, .. } => {
+                count += get_num_statements(body);
+                count += get_num_statements(orelse);
+            }
+            StmtKind::While {
+                test: _,
+                body,
+                orelse,
+            } => {
+                count += 1;
+                count += get_num_statements(body);
+                count += get_num_statements(orelse);
+            }
+            StmtKind::Try {
+                body,
+                handlers,
+                orelse,
+                finalbody,
+            } => {
+                count += 1;
+                count += get_num_statements(body);
+                if !orelse.is_empty() {
+                    count += 1 + get_num_statements(orelse);
+                }
+                if !finalbody.is_empty() {
+                    // weird, but in making samples for pylint, the finally statement counts as 2
```

Comment:
Did you review the Pylint source when writing this? Or was it all inferred from the tests?

Reply:
I made a load of examples by hand and ran pylint on them to see what was going on.

Reply:
Hm, the architecture for pylint is slightly different however:
```
def visit_tryfinally(self, node: nodes.TryFinally) -> None:
    """Increments the branches counter."""
    self._inc_branch(node, 2)
    self._inc_all_stmts(2)
```